Sources/Testing/Testing.docc/testing-for-errors-in-swift-code.md

```diff
@@ -0,0 +1,131 @@
+# Testing for errors in Swift code
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+Ensure that your code handles errors in the way you expect.
+
+## Overview
+
+The Swift language provides an idiomatic approach to [error
+handling](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/errorhandling),
+based on throwing errors where your code detects a failure for a
+caller to catch and react to.
+
+Write tests for your code that validate the conditions in which the
+code throws errors, and the conditions in which it returns without
+throwing an error.  Use overrides of the `#expect()` and
+``require(_:_:sourceLocation:)-5l63q`` macros that check for errors.
+
+### Validate that your code throws an expected error
+
+The Swift structure in this example represents a list that accepts any
+number of toppings for pizzas in the list.  The API contains a method for
+applying a topping to a range of pizzas, and a method for retrieving the
+toppings requested for the item at a given index.  Both of these methods
+throw errors if their parameters are outside the list's range.
+
+```swift
+enum PizzaBase {
+    case deepCrust
+    case shallowCrust
+    case calzone
+}
+
+enum Topping {
+    case tomato
+    case cheese
+    case caper
+    case anchovy
+    case prosciutto
+    case pineapple
+}
+
+struct PizzaToppings {
+    enum PizzaToppingsError : Error {
+        case outOfRange
+    }
+
+    let pizzas: [PizzaBase]
+    var toppings: [Int: [Topping]]
+
+    init(bases: [PizzaBase]) {
+        pizzas = bases
+        toppings = [Int: [Topping]]()
+    }
+
+    mutating func add(topping: Topping, toPizzasIn range: Range<Int>) throws {
+        guard Int(range.startIndex) >= 0 && Int(range.endIndex) < pizzas.count else {
```

Comment:
This is awkward phrasing. What are you trying to do here? We can probably simplify it.

Reply:
I'm trying to test that the range is within the domain of indices in the `pizzas` array, but given that you go on below to say that I don't need to show the code under test, I'll avoid fixing this (which isn't a complete check anyway).